app/test/vesting_test.go

```diff
@@ -0,0 +1,936 @@
+
+		// Stop testing once we hit an account with no spendable balance
+		if vestedCoins.IsZero() {
+			break
+		}
+		s.T().Log("waiting 3 seconds...")
+		time.Sleep(3 * time.Second)
+	}
+}
+
+func (s *VestingModuleTestSuite) TestGenesisContinuousVestingAccountsDelegation() {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find and test a vesting account that has some vesting (locked) balance
+	for {
+		name := s.unusedAccount(ContinuousVestingAccountType)
+		address := getAddress(name, s.cctx.Keyring).String()
+
+		resAccBytes, err := GetRawAccountInfo(s.cctx.GRPCClient, address)
+		assert.NoError(s.T(), err)
+
+		var vAcc vestingtypes.ContinuousVestingAccount
+		err = vAcc.Unmarshal(resAccBytes)
+		assert.NoError(s.T(), err)
+
+		// 10 seconds is chosen to be on the safe side
+		if vAcc.StartTime > tmtime.Now().Unix()+10 {
+			s.testDelegatingVestingAmount(name)
+			return
+		}
+	}
+}
+
+// This test function tests delegation of a continuous vesting account that
+// has part of its allocation unlocked and part of it locked
+func (s *VestingModuleTestSuite) TestGenesisContinuousVestingAccountsDelegationPartiallyVested() {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find and test a vesting account that has some vesting (locked) and
+	// some vested (unlocked) balance
+	for {
+		name := s.unusedAccount(ContinuousVestingAccountType)
+		address := getAddress(name, s.cctx.Keyring).String()
+
+		resAccBytes, err := GetRawAccountInfo(s.cctx.GRPCClient, address)
+		assert.NoError(s.T(), err)
+
+		var vAcc vestingtypes.ContinuousVestingAccount
+		err = vAcc.Unmarshal(resAccBytes)
+		assert.NoError(s.T(), err)
+
+		if vAcc.StartTime < tmtime.Now().Unix() {
+			continue
+		}
+
+		// an account that has its start time passed by just a little
+		if vAcc.StartTime >= tmtime.Now().Unix() {
+			waitTime := vAcc.StartTime - tmtime.Now().Unix() + 1
+			s.T().Logf("waiting %d seconds...", waitTime)
+			time.Sleep(time.Duration(waitTime) * time.Second)
+		}
+		assert.False(s.T(), vAcc.GetVestedCoins(tmtime.Now()).IsZero(), "unlocked balance amount must not be zero")
+
+		s.testDelegatingVestingAmount(name)
+		return
+	}
+}
+
+func (s *VestingModuleTestSuite) TestGenesisContinuousVestingAccountsClaimDelegationRewards() {
+	assert.NoError(s.T(), s.cctx.WaitForNextBlock())
+
+	// find and test a vesting account that has some vesting (locked) balance
+	// to be on the safe side we select one that starts unlocking in at least 20 seconds
+	for {
+		name := s.unusedAccount(ContinuousVestingAccountType)
+		address := getAddress(name, s.cctx.Keyring).String()
+
+		resAccBytes, err := GetRawAccountInfo(s.cctx.GRPCClient, address)
+		assert.NoError(s.T(), err)
+
+		var vAcc vestingtypes.ContinuousVestingAccount
+		err = vAcc.Unmarshal(resAccBytes)
+		assert.NoError(s.T(), err)
+
+		if vAcc.StartTime > tmtime.Now().Unix()+20 {
+			s.testDelegatingVestingAmount(name)
+			s.testClaimDelegationReward(name)
+			return
+		}
+	}
+}
```

Comment:
what are the differences between each of these tests? is there overlap between what is being tested? if so, is there a way to remove any overlap and reduce the cognitive load on readers?

are we testing everything in `TestGenesisContinuousVestingAccountsClaimDelegationRewards`?

Reply:
We can certainly move the overlapped code to a new function

Reply:
Done